src/www/ui_components/server_connection_viz.ts

```diff
@@ -242,62 +231,61 @@ Polymer({
       <img id="large-zero" src\$="[[rootPath]]assets/disc_empty.png" class\$="{{state}}">
       <img id="large" src\$="[[rootPath]]assets/disc_color.png" class\$="green {{animationState}}">
     </div>
-`,
+  `;
 
-  is: 'server-connection-viz',
+  animationStartMS: number;
 
-  properties: {
-    rootPath: String,
-    state: {
-      type: String,
-      observer: '_onStateChanged',
-    },
-    animationState: {
-      type: String,
-      notify: true,
-      value: 'ZERO_STATE',
-    },
-    expanded: {
-      type: Boolean,
-      value: false,
-    },
-  },
+  @property({type: String}) rootPath: string;
+  @property({type: Boolean}) expanded: boolean;
 
-  _onStateChanged: function _onStateChanged(newState) {
-    this._updateAnimationState();
-  },
+  @property({type: String, observer: 'syncAnimationState'}) state: ServerConnectionState;
+  @property({type: String}) animationState: ServerConnectionState = ServerConnectionState.INITIAL;
 
-  // Update CSS when modifying the animation duration.
-  _ANIMATION_DURATION_MS: 1750,
+  @computed('expanded')
+  get expandedClassName() {
+    return this.expanded ? 'expanded' : '';
+  }
 
-  _animationStartMs: null,
+  @computed('state')
+  get shouldAnimate() {
+    return this.isAnimationState(this.state);
+  }
+
+  @computed('animationState')
+  get isAnimating() {
+    return this.isAnimationState(this.animationState);
+  }
 
-  _updateAnimationState: function() {
-    if (this._isAnimating(this.state)) {
-      this._animationStartMs = new Date().getTime();
+  protected syncAnimationState() {
```

Comment:
Why is this protected?

Reply:
If I mark it as private typescript complains that it's "unused" even though it's being used in the template

Reply:
Can we disable the warning somehow (e.g. `@ts-ignore`) and make it private?